locator/src/org_cell_fetching.rs

```diff
@@ -0,0 +1,218 @@
+        let signature = self.generate_signature(&body_json);
+
+        let response = self
+            .client
+            .post(&url)
+            .header("Authorization", format!("rpcsignature {signature}"))
+            .header("Content-Type", "application/json")
+            .body(body_json)
+            .send()
+            .await
+            .with_context(|| format!("Failed to fetch from {url}"))?;
+
+        if !response.status().is_success() {
+            return Err(anyhow::anyhow!(
+                "HTTP error: {} for URL: {}",
+                response.status(),
+                url
+            ));
+        }
+
+        let cell_records = response
+            .json::<CellRecords>()
+            .await
+            .context("Failed to parse response as JSON")?;
+
+        Ok(cell_records)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::time::UNIX_EPOCH;
+    use wiremock::matchers::{body_string, method, path};
+    use wiremock::{Mock, MockServer, ResponseTemplate};
+
+    // HTTP Integration tests using HMAC authentication
+    #[tokio::test]
+    async fn test_fetch_all_success() {
+        let mock_server = MockServer::start().await;
+
+        let response_body = r#"{
+            "records": [
+                {
+                    "org_id": 12345,
+                    "org_name": "some_org",
+                    "cell": "us-1"
+                },
+                {
+                    "org_id": 54321,
+                    "org_name": "some_other_org",
+                    "cell": "de-1"
+                }
+            ]
+        }"#;
+
+        // Expect POST request with empty args JSON body
+        Mock::given(method("POST"))
+            .and(path("/api/0/synapse-rpc/get_cell_records"))
+            .and(body_string(r#"{"args":{}}"#))
+            .respond_with(ResponseTemplate::new(200).set_body_string(response_body))
+            .mount(&mock_server)
+            .await;
+
+        let fetcher = FetchConfig::new(&mock_server.uri(), "test-secret");
+        let result = fetcher.fetch_all().await.unwrap();
+        assert_eq!(result.records.len(), 2);
+        assert_eq!(result.records[0].org_id, 12345);
+        assert_eq!(result.records[0].org_name, "some_org");
+        assert_eq!(result.records[1].org_id, 54321);
+        assert_eq!(result.records[1].org_name, "some_other_org");
+    }
+
+    #[tokio::test]
+    async fn test_fetch_since_with_timestamp() {
+        let mock_server = MockServer::start().await;
+
+        let timestamp = 1640995200u64;
+        let response_body = r#"{
+            "records": [
+                {
+                    "org_id": 999,
+                    "org_name": "blah_blah",
+                    "cell": "us-1"
+                }
+            ]
+        }"#;
+
+        // Expect POST request with timestamp in args JSON body
+        let expected_body = format!(r#"{{"args":{{"since":{timestamp}}}}}"#);
```

Comment:
is it expected that the user will be able to pass `"since": 0` and get every record this way

or is there a maximum retention for this `since` api, and getsentry would return an error that is exceeded?